File: Winter/std::sort is better anways/2.27 class/binarysearch.cpp
#include <iostream>
#include <vector>
using namespace std;
int sign(int number){
    if (number != 0){
        return number/abs(number);
    } else {
        return 0;
    }
}
int binarysearch(vector<int> array, int element, int lowerbound, int upperbound){
    int ourindex = (lowerbound + upperbound)/2;
    if (lowerbound == upperbound){
        if 
    }
    // process 2, no recursion
        // int thesize = array.size();
        // int index = thesize/2;
        // int scale = 4;
        // int indexprevious = thesize + 10;
        // while (true){
        //     int oursign = sign(array[index] - element);
        //     index += ((thesize + oursign * (thesize % 2))/scale) * oursign;
        //     if (index == indexprevious){
        //         if (array[index] == element){
        //             return index;
        //         } else {
        //             return thesize;
        //         }
        //     }
        //     indexprevious = index;
        // }
}
int main(){
    
}

File: Winter/std::sort is better anways/2.27 class/binarysearch.py
def binary(lowerbound, upperbound, array, element)
    if lowerbound == upperbound:
        if array[lowerbound] == element:
            return lowerbound
        else:
            return array.size()
    else:
        number = (lowerbound + upperbound) // 2 
        if array[number] < element:
            return binary(lowerbound, number, array, element)
        else:
            return binary(number, upperbound, array, element)

